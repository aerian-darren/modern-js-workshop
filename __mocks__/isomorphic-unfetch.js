export const emoji = [
    { code: "dog", moji: "🐶", base: "dog" },
    { code: "sign of the horns", moji: "🤘", base: "sign_of_the_horns" },
    { code: "sign of the horns(p)", moji: "🤘🏼", base: "sign_of_the_horns" },
    { code: "sign of the horns(ye)", moji: "🤘🏽", base: "sign_of_the_horns" },
    { code: "sign of the horns(br)", moji: "🤘🏾", base: "sign_of_the_horns" },
    { code: "sign of the horns(bk)", moji: "🤘🏿", base: "sign_of_the_horns" },
    { code: "sign of the horns(wh)", moji: "🤘🏻", base: "sign_of_the_horns" },
    { code: "guardsman(wh)", moji: "💂🏻", base: "guardsman" },
    { code: "guardsman(p)", moji: "💂🏼", base: "guardsman" },
    { code: "guardsman(ye)", moji: "💂🏽", base: "guardsman" },
    { code: "guardsman(br)", moji: "💂🏾", base: "guardsman" },
    { code: "guardsman(bk)", moji: "💂🏿", base: "guardsman" },
    { code: "guardsman", moji: "💂", base: "guardsman" },
];

export const mockFetch = jest.fn().mockReturnValue({
    json: jest.fn().mockReturnValue(emoji),
});

export default mockFetch;

global.fetch = mockFetch;
