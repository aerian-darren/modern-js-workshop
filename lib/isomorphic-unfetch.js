export const emoji = 
    [
        { "code": "dog", "moji": "🐶", "base": "dog" }, 
        { "code": "sign of the horns", "moji": "🤘", "base": "sign_of_the_horns"},
        { "code": "sign of the horns(p)", "moji": "🤘🏼", "base": "sign_of_the_horns"},
        { "code": "sign of the horns(ye)", "moji": "🤘🏽", "base": "sign_of_the_horns"},
        { "code": "sign of the horns(br)", "moji": "🤘🏾", "base": "sign_of_the_horns"},
        { "code": "sign of the horns(bk)", "moji": "🤘🏿", "base": "sign_of_the_horns"},
        { "code": "sign of the horns(wh)", "moji": "🤘🏻", "base": "sign_of_the_horns"}
    ]


export const mockJson = jest.fn().mockReturnValue(emoji);

export const mockFetch = jest.fn().mockReturnValue({
    json: mockJson
});

export default mockFetch;

global.fetch = mockFetch;