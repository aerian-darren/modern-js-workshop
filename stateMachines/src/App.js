import React from "react";
import { Action, State, withStateMachine } from "react-automata";
import LoginForm from "./components/LoginForm";
import { MessageOverlay } from "./components/MessageOverlay";
// import { loginFlow } from "./loginFlow";

const loginFlow = {
    initial: "LoggedOut",
    states: {
        LoggedOut: {
            onEntry: "showLoginForm",
            on: {
                LOGIN: "LoggingIn",
            },
        },
        LoggingIn: {
            onEntry: "showLoggingInMessage",
            on: {
                ERROR: "Error",
                SUCCESS: "Success",
            },
        },
        Success: {
            onEntry: "showSuccessMessage",
            on: {
                TIMEOUT: "LoggedIn",
            },
        },
        Error: {
            onEntry: "showErrorMessage",
            on: {
                TIMEOUT: "LoggedOut",
            },
        },
        LoggedIn: {
            onEntry: "showLoggedIn",
            on: {
                LOGOUT: "LoggedOut",
            },
        },
    },
};

const mockUserDetails = {
    username: "aerian",
    password: "isCool,reallyCool",
};

const messages = {
    error: "There was an error with your username or password",
    success: "You have been logged in",
    info: "Logging in...",
};

const DEFAULT_STATE = {
    message: "",
};

export class App extends React.Component {
    state = DEFAULT_STATE;

    mockLoginAPI = async (username, password) => {
        await window.setTimeout(() => {
            if (
                username !== mockUserDetails.username ||
                password !== mockUserDetails.password
            ) {
                this.props.transition("ERROR");
            } else {
                this.props.transition("SUCCESS");
            }
        }, 5000);
    };

    showLoggingInMessage() {
        this.setState({ message: "info" });
    }

    showErrorMessage() {
        this.setState({ message: "error" });
    }

    showSuccessMessage() {
        this.setState({ message: "success" });
    }

    componentWillTransition(event) {
        console.log({ event });
    }

    handleMessageClose = () => {
        this.props.transition("TIMEOUT");
    };

    handleClick = ({ username, password }) => {
        this.props.transition("LOGIN");
        this.mockLoginAPI(username, password);
    };

    render() {
        return (
            <main>
                <p>
                    {this.props.machineState && this.props.machineState.value}
                </p>
                <State is={["LoggedOut", "LoggingIn", "Error", "Success"]}>
                    <LoginForm login={this.handleClick} />
                </State>
                <Action is="showLoggingInMessage">
                    <MessageOverlay messageType={this.state.message}>
                        {messages[this.state.message]}
                    </MessageOverlay>
                </Action>
                <Action is={["showErrorMessage", "showSuccessMessage"]}>
                    <MessageOverlay
                        onClosed={this.handleMessageClose}
                        timeout={true}
                        messageType={this.state.message}
                    >
                        {messages[this.state.message]}
                    </MessageOverlay>
                </Action>
            </main>
        );
    }
}

export default withStateMachine(loginFlow)(App);
