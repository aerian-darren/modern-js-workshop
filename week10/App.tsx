import * as React from "react";
import { EmojiDisplay } from "./EmojiDisplay";

import emojilist from "../lib/emoji";
import { EmojiKeyboard } from "./EmojiKeyboard";

interface State {
    output: string;
}

interface Props {}

const DEFAULT_STATE: State = {
    output: "Type your emoji",
};
export class App extends React.Component<Props, State> {
    public state = DEFAULT_STATE;

    public addEmoji = (moji: string) => {
        let output = moji;
        if (this.state.output !== DEFAULT_STATE.output) {
            output = this.state.output + moji;
        }
        this.setState({ output });
    };

    public render() {
        return (
            <div className="typewriter">
                <EmojiDisplay>{this.state.output}</EmojiDisplay>
                <EmojiKeyboard emojis={emojilist} onAddEmoji={this.addEmoji} />
            </div>
        );
    }
}
