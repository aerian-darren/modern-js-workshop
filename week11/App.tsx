import * as React from "react";
import { EmojiDisplay } from "./EmojiDisplay";

import emojilist from "../lib/emoji";
import {
    fetchEmojis,
    getCategories,
    makeAFilterByCategory,
} from "../lib/emojilib";
import { CategorySelector } from "./CategorySelector";
import { Emoji } from "./Emoji";
import { EmojiKeyboard } from "./EmojiKeyboard";
interface State {
    output: string[];
    emojilist: Emoji[];
    categories: string[];
    filteredEmojis: Emoji[];
    selectedCategory: string;
}

interface Props {}

const DEFAULT_CATEGORIES = getCategories(emojilist);
const DEFAULT_CATEGORY = DEFAULT_CATEGORIES[1];

const DEFAULT_STATE: State = {
    output: ["Type your emoji"],
    categories: DEFAULT_CATEGORIES,
    selectedCategory: DEFAULT_CATEGORY,
    filteredEmojis: emojilist.filter(makeAFilterByCategory(DEFAULT_CATEGORY)),
    emojilist,
};

export class App extends React.Component<Props, State> {
    public readonly state = DEFAULT_STATE;

    public componentDidMount() {
        this.loadRemoteEmojis();
    }

    public loadRemoteEmojis = async () => {
        const emojis = await fetchEmojis();
        this.setState(
            { emojilist: emojis, categories: getCategories(emojis) },
            () => this.updateCategory(this.state.categories[0])
        );
    };

    public clear = () => this.setState({ output: DEFAULT_STATE.output });
    public addEmoji = (moji: string) => {
        let output = [moji];
        if (this.state.output !== DEFAULT_STATE.output) {
            output = [...this.state.output, moji];
        }
        this.setState({ output });
    };

    public removeEndEmoji = () => {
        if (this.state.output.length > 1) {
            return this.setState({
                output: this.state.output.splice(
                    0,
                    this.state.output.length - 1
                ),
            });
        }
        return this.setState({ output: DEFAULT_STATE.output });
    };

    public updateCategory = (selectedCategory: string) => {
        this.setState({
            selectedCategory,
            filteredEmojis: this.state.emojilist.filter(
                makeAFilterByCategory(selectedCategory)
            ),
        });
    };

    public changeCategory = () => this.updateCategory(this.state.categories[3]);

    public onCategoryChange = (event: React.FormEvent<HTMLSelectElement>) =>
        this.updateCategory(event.currentTarget.value);

    public render() {
        return (
            <div className="typewriter">
                <EmojiDisplay>{this.state.output}</EmojiDisplay>
                <CategorySelector
                    categories={this.state.categories}
                    onChange={this.onCategoryChange}
                    value={this.state.selectedCategory}
                />
                <button onClick={this.clear}>Clear</button>
                <button onClick={this.removeEndEmoji}>Del</button>
                <button onClick={this.changeCategory}>Change it</button>
                <EmojiKeyboard
                    emojis={this.state.filteredEmojis}
                    onAddEmoji={this.addEmoji}
                    clickedKeys={this.state.output}
                />
                <EmojiDisplay>{this.state.output}</EmojiDisplay>
            </div>
        );
    }
}
