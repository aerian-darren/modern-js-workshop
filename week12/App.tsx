import * as React from "react";
import emojilist from "../lib/emoji";
import {
    fetchEmojis,
    getCategories,
    makeAFilterByCategory,
} from "../lib/emojilib";
import { CategorySelector } from "./CategorySelector";
import { Emoji } from "./Emoji";
import { EmojiDisplay } from "./EmojiDisplay";
import { EmojiKeyboard } from "./EmojiKeyboard";
import { Key } from "./Key";
import { withRedBorder } from "./withRedBorder";

const RedKey = withRedBorder(Key);

interface State {
    output: string[];
    currentEmojilist: Emoji[];
    categories: string[];
    filteredEmojis: Emoji[];
    selectedCategory: string;
}

interface Props {}

const DEFAULT_CATEGORIES: string[] = getCategories(emojilist);
const DEFAULT_CATEGORY = DEFAULT_CATEGORIES[0];

let remoteEmojilist: Emoji[] = [];

const DEFAULT_STATE: State = {
    output: ["Type your emoji"],
    categories: DEFAULT_CATEGORIES,
    selectedCategory: DEFAULT_CATEGORY,
    filteredEmojis: emojilist.filter(makeAFilterByCategory(DEFAULT_CATEGORY)),
    currentEmojilist: emojilist,
};

export class App extends React.Component<Props, State> {
    public readonly state = DEFAULT_STATE;

    public toggleEmojilist = () => {
        if (!remoteEmojilist.length) {
            return this.loadRemoteEmojis();
        }

        const currentEmojilist =
            this.state.currentEmojilist === emojilist
                ? remoteEmojilist
                : emojilist;
        return this.setState(
            {
                currentEmojilist,
                categories: getCategories(currentEmojilist),
            },
            () => this.updateCategory(this.state.selectedCategory)
        );
    };

    public loadRemoteEmojis = async () => {
        remoteEmojilist = await fetchEmojis();
    };

    public clear = () => this.setState({ output: DEFAULT_STATE.output });
    public addEmoji = (moji: string) => {
        let output = [moji];
        if (this.state.output !== DEFAULT_STATE.output) {
            output = [...this.state.output, moji];
        }
        this.setState({ output });
    };

    public removeEndEmoji = () => {
        if (this.state.output.length > 1) {
            return this.setState({
                output: this.state.output.splice(
                    0,
                    this.state.output.length - 1
                ),
            });
        }
        return this.setState({ output: DEFAULT_STATE.output });
    };

    public updateCategory = (selectedCategory: string) => {
        this.setState({
            selectedCategory,
            filteredEmojis: this.state.currentEmojilist.filter(
                makeAFilterByCategory(selectedCategory)
            ),
        });
    };

    public changeCategory = () => this.updateCategory(this.state.categories[3]);

    public onCategoryChange = (event: React.FormEvent<HTMLSelectElement>) =>
        this.updateCategory(event.currentTarget.value);

    public render() {
        return (
            <div className="typewriter">
                <EmojiDisplay>{this.state.output}</EmojiDisplay>
                <CategorySelector
                    categories={this.state.categories}
                    onChange={this.onCategoryChange}
                    value={this.state.selectedCategory}
                />
                <RedKey onClick={this.clear} label="Clear" />
                <button onClick={this.removeEndEmoji}>⬅️ Delete</button>
                <button onClick={this.toggleEmojilist}>
                    {!remoteEmojilist.length
                        ? "Load more!"
                        : this.state.currentEmojilist === emojilist
                            ? "use remote emojis"
                            : "use local emojis"}
                </button>
                <EmojiKeyboard
                    emojis={this.state.filteredEmojis}
                    onAddEmoji={this.addEmoji}
                    clickedKeys={this.state.output}
                />
                <EmojiDisplay>{this.state.output}</EmojiDisplay>
            </div>
        );
    }
}
