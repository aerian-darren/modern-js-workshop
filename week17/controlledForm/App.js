import React, { Component } from "react";
import Form from './components/form';
import "./styles.css";

class App extends Component {
    render() {
        return (
        <div className="App">
            <Form />
        </div>
        )
    }
}

export default App;
