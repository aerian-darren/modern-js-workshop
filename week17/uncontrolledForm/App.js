import React, { Component } from "react";
import _ from 'lodash';
// import "./styles.css";

export class App extends Component {

    constructor (props) {
        super(props);
        this.fields = {};

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    gatherData () {
        return _.mapValues(this.fields, 'value');
    }

    handleSubmit (e) {
        e.preventDefault();
        console.table(this.gatherData());
    }

    renderField(filedText, filedId) {
        return (
            <div className="field">
                <label className="field__label" htmlFor={ filedId }>{ filedText }</label>
                <input 
                    id ={ filedId } 
                    required 
                    type="text"
                    maxLength="50"
                    className="field__input field__input--text"
                    ref={ (e) => this.fields.postcode = e } />
            </div>
        )
    }

    renderFooter() {
        return (
            <footer className="footer">
                <span className="footer__error">Passwords do not match</span>
                <button className="footer__button" disabled type="submit">Submit</button>
            </footer>
        )
    }

    render() {
        return (
            <div className="conatiner">
                <form id="form" className="form" onSubmit={this.handleSubmit}>
                    <h1 className="form__header">Would you like to know more?</h1>
                    <fieldset className="form_fieldset">
                        { this.renderField("Username", "username") }
                        { this.renderField("Email", "email") }
                        { this.renderField("Password", "password1") }
                        { this.renderField("Please confirm password", "password2") }
                    </fieldset>
                    { this.renderFooter() }
                </form>
            </div>
        );
    }
}
