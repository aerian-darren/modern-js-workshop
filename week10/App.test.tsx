import "isomorphic-unfetch";
import * as React from "react";
import * as ReactDOM from "react-dom";
import * as renderer from "react-test-renderer";
import { App } from "./App";

// it("renders without crashing", () => {
//     const run = () => {
//         const div = document.createElement("div");
//         ReactDOM.render(<App />, div);
//         ReactDOM.unmountComponentAtNode(div);
//     };
//     expect(run).not.toThrow();
// });

it("renders all the components we need", () => {
    const app = renderer.create(<App />);
    const instance = app.getInstance();
    if (instance) {
        instance.componentDidMount = jest.fn();
    }
    const tree = app.toJSON();
    let treeChildrenLength = 0;

    if (tree && tree.children) {
        treeChildrenLength = tree.children.length;
    }

    expect(treeChildrenLength).toBeGreaterThanOrEqual(2);
    expect(tree).toMatchSnapshot();
});
