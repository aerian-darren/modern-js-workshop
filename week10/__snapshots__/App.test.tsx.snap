// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all the components we need 1`] = `
<div
  className="typewriter"
>
  <h1>
    Type your emoji
  </h1>
  <select
    onChange={[Function]}
  >
    <option
      value="objects"
    >
      Objects
    </option>
    <option
      value="faces"
    >
      Faces
    </option>
    <option
      value="gestures"
    >
      Gestures
    </option>
    <option
      value="transportation"
    >
      Transportation
    </option>
    <option
      value="people"
    >
      People
    </option>
  </select>
  <button
    onClick={[Function]}
  >
    Clear
  </button>
  <div
    id="keyboard"
  >
    <button
      onClick={[Function]}
    >
      🎿
    </button>
    <button
      onClick={[Function]}
    >
      🏷
    </button>
    <button
      onClick={[Function]}
    >
      👑
    </button>
    <button
      onClick={[Function]}
    >
      👕
    </button>
    <button
      onClick={[Function]}
    >
      👖
    </button>
    <button
      onClick={[Function]}
    >
      👗
    </button>
    <button
      onClick={[Function]}
    >
      👛
    </button>
    <button
      onClick={[Function]}
    >
      👝
    </button>
    <button
      onClick={[Function]}
    >
      👢
    </button>
    <button
      onClick={[Function]}
    >
      💊
    </button>
    <button
      onClick={[Function]}
    >
      💍
    </button>
    <button
      onClick={[Function]}
    >
      💎
    </button>
    <button
      onClick={[Function]}
    >
      💴
    </button>
    <button
      onClick={[Function]}
    >
      💶
    </button>
    <button
      onClick={[Function]}
    >
      💷
    </button>
    <button
      onClick={[Function]}
    >
      💺
    </button>
    <button
      onClick={[Function]}
    >
      📀
    </button>
    <button
      onClick={[Function]}
    >
      📅
    </button>
    <button
      onClick={[Function]}
    >
      📔
    </button>
    <button
      onClick={[Function]}
    >
      📖
    </button>
    <button
      onClick={[Function]}
    >
      📚
    </button>
    <button
      onClick={[Function]}
    >
      📟
    </button>
    <button
      onClick={[Function]}
    >
      📠
    </button>
    <button
      onClick={[Function]}
    >
      📣
    </button>
    <button
      onClick={[Function]}
    >
      📻
    </button>
    <button
      onClick={[Function]}
    >
      📼
    </button>
    <button
      onClick={[Function]}
    >
      🔑
    </button>
    <button
      onClick={[Function]}
    >
      🔒
    </button>
    <button
      onClick={[Function]}
    >
      🔔
    </button>
    <button
      onClick={[Function]}
    >
      🕳
    </button>
    <button
      onClick={[Function]}
    >
      🖪
    </button>
    <button
      onClick={[Function]}
    >
      🖫
    </button>
    <button
      onClick={[Function]}
    >
      🖺
    </button>
    <button
      onClick={[Function]}
    >
      🗈
    </button>
    <button
      onClick={[Function]}
    >
      🗏
    </button>
    <button
      onClick={[Function]}
    >
      🗐
    </button>
    <button
      onClick={[Function]}
    >
      🗿
    </button>
    <button
      onClick={[Function]}
    >
      🚪
    </button>
    <button
      onClick={[Function]}
    >
      🛀
    </button>
    <button
      onClick={[Function]}
    >
      ⌚️
    </button>
  </div>
  <h1>
    Type your emoji
  </h1>
</div>
`;
